### PR TITLE
chore(workflow): add semantic pull request check

### PR DESCRIPTION
**Pull request by 0618**
#### Description of changes

Add [Semantic Pull Request](https://github.com/marketplace/actions/semantic-pull-request) check to enforce the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)

#### Checklist

- [x] PR description included
- [ ] `yarn test` passes
- [ ] Tests are updated
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by slaymance — APPROVED** [review]
Looks great! With this workflow, the changesets PR won't pass the convention. It's probably easiest to rename the PR created to `chore(release): Version Packages`.

File: .github/workflows/semantic-pull-request.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+      - opened
+      - edited
+      - synchronize
+
+jobs:
+  main:
+    name: Validate PR title
+    runs-on: ubuntu-latest
+    steps:
+      - uses: amannn/action-semantic-pull-request@v4
```
Comment (zchenwei) [inline .github/workflows/semantic-pull-request.yml:15]: This will need security review

File: .github/workflows/semantic-pull-request.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+name: 'Lint PR'
+
+on:
+  pull_request_target:
```
Comment (zchenwei) [inline .github/workflows/semantic-pull-request.yml:4]: `pull_request_target` will expose token access to actions triggered by a fork. We gotta be very careful and this is what @wlee221 did to mitigate it <- https://github.com/aws-amplify/amplify-ui/pull/855

File: .github/workflows/semantic-pull-request.yml (hunk 1 above)
Comment (zchenwei) [reply .github/workflows/semantic-pull-request.yml:15]: This is settings about how we restricted actions on Github. Changeset has been signed off by AppSec <- https://github.com/aws-amplify/amplify-ui/settings/actions

File: .github/workflows/semantic-pull-request.yml (hunk 2 above)
Comment (wlee221) [reply .github/workflows/semantic-pull-request.yml:4]: Oh man, you found my fossil 😮

Comment (wlee221) [reply .github/workflows/semantic-pull-request.yml:4]: This is fine, as long as you don't checkout any of the PR content:

> TL;DR: Combining pull_request_target workflow trigger with an explicit checkout of an untrusted PR is a dangerous practice that may lead to repository compromise. <br /> <br />  https://securitylab.github.com/research/github-actions-preventing-pwn-requests/

**Review by reesscot — CHANGES_REQUESTED** [review]
Thanks for putting this together MJ! Just adding merge blocker since this requires AppSec review. Let's work on this later in the month

**Comment by calebpollman** [comment]
Can we close this? @0618  @reesscot

**Pull request closed without merging**